### PR TITLE
Adding documentation for deltaproxy.

### LINKED ISSUE #61220: [DOCS] Deltaproxy documentation missing
**Description**
Deltaproxy documentation missing

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding documentation for deltaproxy.

### What issues does this PR fix or reference?
Fixes: #61220

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: changelog/61220.added [hunk 1]
```diff
@@ -0,0 +1 @@
+Adding documentation for deltaproxy.
```
Comment (Ch3LL) [inline changelog/61220.added:1]: changelogs are not needed for documentation changes.

File: doc/ref/configuration/delta_proxy.rst [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+have ideally already tested and used standard Salt proxy minions in their
+environment before deciding to move to a delta proxy minion environment. See
+`Salt proxy minions
+<https://docs.saltstack.com/en/latest/topics/proxyminion/index.html>`_ for more
+information.
+
+.. Note::
+    If you have not used standard Salt proxy minions before, consider testing
+    and deploying standard Salt proxy minions in your environment first.
+    Alternatively, you can request consulting services for assistance.
```
Comment (Ch3LL) [inline doc/ref/configuration/delta_proxy.rst:49]:
```suggestion
    Alternatively, you can request consulting services for assistance.
```

Is this required? How would a user know how to contact consulting services?

File: doc/ref/configuration/delta_proxy.rst [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+    * `Salt proxy minions
+      <https://docs.saltstack.com/en/latest/topics/proxyminion/index.html>`_
+
+    * `Salt proxy modules
+      <https://docs.saltstack.com/en/latest/ref/proxy/all/index.html#all-salt-proxy>`_
+
+
+When Delta Proxy minions are needed
+-------------------------------------
```
Comment (Ch3LL) [inline doc/ref/configuration/delta_proxy.rst:87]: looks like a couple extra `-`

File: doc/ref/configuration/delta_proxy.rst [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+      pillar file. Pillars allow confidential, targeted data to be securely sent
+      only to the relevant minion. Because all configurations for
+      Delta Proxy minions are done on the Salt master (not on the minions), you
+      use pillar files to configure the deltaproxy-minion service.
+
+  * - top file
+    - The top file is a pillar file that maps which states should be applied to
+      different minions in certain environments.
+
+    .. _delta-proxy-install:
```
Comment (Ch3LL) [inline doc/ref/configuration/delta_proxy.rst:157]: I *think* this needs to be moved back a couple spaces but not certain.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
we should also add this to the freeze branch so we can create a new docs tag so these docs show up in the 3004 release.

**Review by barbaricyawps — CHANGES_REQUESTED** [review]
**Praise:** Hey, thank you for putting these docs together, Gareth. This is a huge weight off my shoulders, personally. I love seeing everyone contributing to the docs and it's a big reason why I'm excited to be on this team.

I have a bunch of minor suggestions. I hope they don't feel overwhelming. Let me know what I can do to help.

File: doc/ref/configuration/delta_proxy.rst [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+
+.. Note::
+    If you have not used standard Salt proxy minions before, consider testing
+    and deploying standard Salt proxy minions in your environment first.
+    Alternatively, you can request consulting services for assistance.
+
+
+Proxy minions vs. Delta Proxy minions
+=========================================
+Salt can target network devices through `Salt proxy minions
```
[the remark this answers is not in the document]
Comment (garethgreenaway) [reply doc/ref/configuration/delta_proxy.rst:26]: I believe this one is resolved but is still showing.

**Review by barbaricyawps — APPROVED** [review]
Looks good!

**Review by Ch3LL — APPROVED** [review]

**Review by cmcmarrow — APPROVED** [review]

**Pull request merged**